### PR TITLE
Jormun: Manage olympic sites

### PR DESCRIPTION
**Pull request by azime**
Jira: [NAV-2019](https://navitia.atlassian.net/browse/NAV-2019)

[NAV-2019]: https://navitia.atlassian.net/browse/NAV-2019?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: source/jormungandr/jormungandr/instance.py [hunk 1]
```diff
@@ -83,6 +82,11 @@
     "calendar": request_pb2.PlaceCodeRequest.Calendar,  # type: ignore
 }
 
+OlympicCriteria = namedtuple(
+    'OlympicCriteria',
```
Comment (pbench) [inline source/jormungandr/jormungandr/instance.py]:
```suggestion
OlympicsForbiddenUris = namedtuple(
    'OlympicsForbiddenUris',
```

File: source/jormungandr/jormungandr/instance.py [hunk 2]
```diff
@@ -83,6 +82,11 @@
...
+    ['pt_object_olympic_uris', 'poi_property_key', 'poi_property_value'],
```
Comment (pbench) [inline source/jormungandr/jormungandr/instance.py]:
```suggestion
    ['pt_object_olympics_forbidden_uris', 'poi_property_key', 'poi_property_value'],
```

File: source/jormungandr/jormungandr/instance.py [hunk 3]
```diff
@@ -106,6 +110,30 @@ def _getter(self):
     return property(_getter)
 
 
+def parse_and_get_olympic_criteria(dict_olympic_criteria):
```
Comment (pbench) [inline source/jormungandr/jormungandr/instance.py]:
```suggestion
def parse_and_get_olympics_forbiddent_uris(dict_olympic_forbidden_uris):
```

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 4]
```diff
@@ -1209,6 +1209,41 @@ def isochrone_common(isochrone, request, instance, journey_req):
...
+                and p.value == instance.olympic_criteria.poi_property_value
+            ),
+            None,
+        )
+        if criteria:
+            return True
+    return False
+
+
+def make_olympic_criteria(origin_detail, destination_detail, api_request, instance):
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/new_default.py]:
```suggestion
def add_olympics_forbidden_uris(origin_detail, destination_detail, api_request, instance):
```

File: source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py [hunk 5]
```diff
@@ -721,3 +724,126 @@ def mock_get_best_boarding_position(from_id, to_id):
...
+            street_network_configurations=[],
+            ridesharing_configurations={},
+            instance_equipment_providers=[],
+            realtime_proxies_configuration=[],
+            pt_planners_configurations={},
+            zmq_socket_type=None,
+            autocomplete_type='kraken',
+            streetnetwork_backend_manager=None,
+            external_service_provider_configurations=[],
+            olympic_criteria=criteria,
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py]:
```suggestion
            olympic_forbidden_uris=olympic_forbidden_uris,
```

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 6]
```diff
@@ -1209,6 +1209,41 @@ def isochrone_common(isochrone, request, instance, journey_req):
...
+        return
+    if is_olympic_site(origin_detail, instance):
+        return
+    if is_olympic_site(destination_detail, instance):
+        return
+
+    if api_request.get("forbidden_uris[]"):
+        api_request["forbidden_uris[]"] += instance.olympic_criteria.pt_object_olympic_uris
+    else:
+        api_request["forbidden_uris[]"] = instance.olympic_criteria.pt_object_olympic_uris
```
Comment (woshilapin) [inline source/jormungandr/jormungandr/scenarios/new_default.py]:
```suggestion
    api_request.setdefault("forbidden_uris[]", [])
    api_request["forbidden_uris[]"] += instance.olympic_criteria.pt_object_olympic_uris
```

**Review by woshilapin — APPROVED** [review]
Nice test suite :tada:

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 6 above)
Comment (azime) [reply source/jormungandr/jormungandr/scenarios/new_default.py]: Done

File: source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py (hunk 5 above)
Comment (azime) [reply source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py]: Done

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 4 above)
Comment (azime) [reply source/jormungandr/jormungandr/scenarios/new_default.py]: Done

File: source/jormungandr/jormungandr/instance.py (hunk 3 above)
Comment (azime) [reply source/jormungandr/jormungandr/instance.py]: Done

File: source/jormungandr/jormungandr/instance.py (hunk 2 above)
Comment (azime) [reply source/jormungandr/jormungandr/instance.py]: Done

File: source/jormungandr/jormungandr/instance.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/jormungandr/instance.py]: Done

**Review by pbench — APPROVED** [review]

**Pull request merged**